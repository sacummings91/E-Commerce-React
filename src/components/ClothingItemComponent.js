import React from 'react';
import { NavLink } from 'react-router-dom';
import { Card, Image, Header, Button } from 'semantic-ui-react';

export default function ClothingItemComponent({ item, addToCart }) {
  console.log(item);
  function onClick(event) {
    // event.preventDefault();
    addToCart(item);
  }

  return (
    <div className="ClothingItemComponent">
      <Card style={{ maxWidth: '220px' }}>
        <Image
          fluid
          src={item.imageUrl}
          alt={item.name}
          as={NavLink}
          exact
          to="/"
        />
        <Card.Content>
          <Header as="h3" textAlign="center">
            <NavLink exact to={`/products/${item.id}`}>
              {item.name || 'N/A'}
            </NavLink>
          </Header>
          <Header as="h3" textAlign="center">
            {item.price ? `$${item.price.toFixed(2)}` : 'N/A'}
          </Header>
          <Button
            as={NavLink}
            exact
            to="/cart"
            className="add-to-cart-button"
            color="black"
            onClick={onClick}>
            Add To Cart
          </Button>
        </Card.Content>
      </Card>
    </div>
  );
}
