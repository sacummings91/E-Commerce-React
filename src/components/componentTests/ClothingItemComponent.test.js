import React from 'react';
import { shallow, mount } from 'enzyme';
import { Image } from 'semantic-ui-react';
import ClothingItemComponent from '../ClothingItemComponent';

const addToCart = jest.fn();

const item = {
  name: 'Black Tee',
  price: 24.99,
  imageUrl: '//via.placeholder.com/264x264'
};

describe('ClothingItemComponent shallowWrapper test', () => {
  const shallowWrapper = shallow(
    <ClothingItemComponent addToCart={addToCart} item={item} />
  );

  it('should console.log the shallowWrapper object', () => {
    console.log(shallowWrapper.debug());
  });

  it('Should have an add to cart button', () => {
    expect(shallowWrapper.find('.add-to-cart-button')).toHaveLength(1);
  });

  // it('Should have an Image element', () => {
  //   expect(shallowWrapper.contains(<Image as="img" alt={item.name} />)).toBe(
  //     true
  //   );
  // });

  it('Should fire addToCart when button is clicked', () => {
    shallowWrapper.find('.add-to-cart-button').simulate('click', {
      preventDefault: () => {}
    });

    expect(addToCart).toBeCalled();
  });

  it('Should pass snapshot test', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
});

// describe('ClothingItemComponent full render tests', () => {
//   const fullWrapper = mount(
//     <ClothingItemComponent addToCart={addToCart} item={item} />
//   );
//
//   it('Should test item prop', () => {
//     expect(fullWrapper.props().item).toEqual(item);
//   });
// });
