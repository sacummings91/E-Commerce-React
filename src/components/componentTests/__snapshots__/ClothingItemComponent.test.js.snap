// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClothingItemComponent shallowWrapper test Should pass snapshot test 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="ClothingItemComponent"
>
    <Card>
        <Image
            alt="Black Tee"
            as="img"
            src="//via.placeholder.com/264x264"
            ui={true}
        />
        <CardContent>
            <Header
                as="h2"
                textAlign="center"
            >
                <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    exact={true}
                    to="/products/undefined"
                >
                    Black Tee
                </NavLink>
            </Header>
            <Header
                as="h3"
                textAlign="center"
            >
                $24.99
            </Header>
            <Button
                as="button"
                className="add-to-cart-button"
                color="black"
                onClick={[Function]}
            >
                <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="add-to-cart"
                    exact={true}
                    to="/cart"
                >
                    Add To Cart
                </NavLink>
            </Button>
        </CardContent>
    </Card>
</div>,
  "nodes": Array [
    <div
      className="ClothingItemComponent"
>
      <Card>
            <Image
                  alt="Black Tee"
                  as="img"
                  src="//via.placeholder.com/264x264"
                  ui={true}
            />
            <CardContent>
                  <Header
                        as="h2"
                        textAlign="center"
                  >
                        <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              exact={true}
                              to="/products/undefined"
                        >
                              Black Tee
                        </NavLink>
                  </Header>
                  <Header
                        as="h3"
                        textAlign="center"
                  >
                        $24.99
                  </Header>
                  <Button
                        as="button"
                        className="add-to-cart-button"
                        color="black"
                        onClick={[Function]}
                  >
                        <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="add-to-cart"
                              exact={true}
                              to="/cart"
                        >
                              Add To Cart
                        </NavLink>
                  </Button>
            </CardContent>
      </Card>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ClothingItemComponent
        addToCart={[Function]}
        item={
                Object {
                        "imageUrl": "//via.placeholder.com/264x264",
                        "name": "Black Tee",
                        "price": 24.99,
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "addToCart": [Function],
          "item": Object {
            "imageUrl": "//via.placeholder.com/264x264",
            "name": "Black Tee",
            "price": 24.99,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="ClothingItemComponent"
>
          <Card>
                    <Image
                              alt="Black Tee"
                              as="img"
                              src="//via.placeholder.com/264x264"
                              ui={true}
                    />
                    <CardContent>
                              <Header
                                        as="h2"
                                        textAlign="center"
                              >
                                        <NavLink
                                                  activeClassName="active"
                                                  ariaCurrent="true"
                                                  exact={true}
                                                  to="/products/undefined"
                                        >
                                                  Black Tee
                                        </NavLink>
                              </Header>
                              <Header
                                        as="h3"
                                        textAlign="center"
                              >
                                        $24.99
                              </Header>
                              <Button
                                        as="button"
                                        className="add-to-cart-button"
                                        color="black"
                                        onClick={[Function]}
                              >
                                        <NavLink
                                                  activeClassName="active"
                                                  ariaCurrent="true"
                                                  className="add-to-cart"
                                                  exact={true}
                                                  to="/cart"
                                        >
                                                  Add To Cart
                                        </NavLink>
                              </Button>
                    </CardContent>
          </Card>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="ClothingItemComponent"
>
          <Card>
                    <Image
                              alt="Black Tee"
                              as="img"
                              src="//via.placeholder.com/264x264"
                              ui={true}
                    />
                    <CardContent>
                              <Header
                                        as="h2"
                                        textAlign="center"
                              >
                                        <NavLink
                                                  activeClassName="active"
                                                  ariaCurrent="true"
                                                  exact={true}
                                                  to="/products/undefined"
                                        >
                                                  Black Tee
                                        </NavLink>
                              </Header>
                              <Header
                                        as="h3"
                                        textAlign="center"
                              >
                                        $24.99
                              </Header>
                              <Button
                                        as="button"
                                        className="add-to-cart-button"
                                        color="black"
                                        onClick={[Function]}
                              >
                                        <NavLink
                                                  activeClassName="active"
                                                  ariaCurrent="true"
                                                  className="add-to-cart"
                                                  exact={true}
                                                  to="/cart"
                                        >
                                                  Add To Cart
                                        </NavLink>
                              </Button>
                    </CardContent>
          </Card>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ClothingItemComponent
    addToCart={[Function]}
    item={
        Object {
            "imageUrl": "//via.placeholder.com/264x264",
            "name": "Black Tee",
            "price": 24.99,
          }
    }
/>,
}
`;
